gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/LTFDocuments.java

```diff
@@ -0,0 +1,130 @@
+package edu.isi.vista.gigawordIndexer;
+
+import com.google.common.collect.AbstractIterator;
+import com.google.common.collect.ImmutableList;
+import com.google.common.io.ByteSource;
+import edu.isi.nlp.corpora.LctlText;
+import edu.isi.nlp.corpora.LtfDocument;
+import edu.isi.nlp.corpora.LtfReader;
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipFile;
+import org.apache.commons.io.IOUtils;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LTFDocuments implements Iterable<Article>, Closeable {
+
+  private static final Logger log = LoggerFactory.getLogger(LTFDocuments.class);
+
+  private ZipFile zipFile;
+
+  private Enumeration<? extends ZipEntry> ltfZipEntries;
+
+
+  @Override
+  public void close() throws IOException {
+    zipFile.close();
+  }
+
+  @NotNull
+  @Override
+  public Iterator<Article> iterator() {
+    return new ArticlesIterator();
+  }
+
+  private LTFDocuments(ZipFile zipFile) {
+    this.zipFile = zipFile;
+    this.ltfZipEntries = zipFile.entries();
+
+  }
+
+  static LTFDocuments fromLTFZippedFile(Path p) throws IOException {
+
+    ZipFile zipFile = new ZipFile(p.toFile(), StandardCharsets.UTF_8);
+    return new LTFDocuments(zipFile);
+
+  }
+
+
+  private class ArticlesIterator extends AbstractIterator<Article> {
+    private LtfReader reader = new LtfReader();
+    private ZipEntry currentLtfEntry;
+    private ImmutableList<LtfDocument> entryDocs;
+    private int docIndex;
+
+    private ArticlesIterator() {
+      // get first zip entry of Ltf from zip file
+      currentLtfEntry = nextLtfEntry();
+      try {
+        entryDocs = getDocsForLtfEntry(currentLtfEntry);
+        docIndex = 0;
+      } catch (IOException e) {
+        log.error("Error reading zip file entry: {} in {}", currentLtfEntry.getName(), zipFile.getName());
+        log.error(e.getMessage());
+        System.exit(1);
```

Comment:
You almost never want to do `System.exit` except at the top-level of a program.  What if the user doesn't want their whole program to crash just because this had a problem?  You can wrap the checked `IOException` in an unchecked `RuntimeException`:

```
throw RuntimeException(e)
```

Reply:
Fixed.